qemu/tests/msi_change_flag.py

```diff
@@ -0,0 +1,126 @@
+        driver = params.get("driver_name")
+        device_name = params["device_name"]
+        error_context.context("Boot guest with %s device" % driver, logging.info)
+        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                                test, driver,
+                                                                login_timeout)
+        error_context.context("Check %s's irq number" % device_name, logging.info)
+        devcon_folder = utils_misc.set_winutils_letter(session, params["devcon_folder"])
+        hwid = win_dev.get_hwids(session, device_name, devcon_folder, login_timeout)[0]
+        get_irq_cmd = params["get_irq_cmd"] % (devcon_folder, hwid)
+        irq_list = re.findall(r':\s+(\d+)', session.cmd_output(get_irq_cmd), re.M)
+        if not irq_list:
+            test.error("device %s's irq checked fail" % device_name)
+        irq_nums = len(irq_list)
+        for irq_symbol in (ctypes.c_int32(int(irq)).value for irq in irq_list):
+            if not irq_nums > 1 and irq_symbol < 0:
+                test.fail("%s's irq is not correct." % device_name)
+        if params.get("msi_cmd"):
+            error_context.context("Disable MSI in guest", logging.info)
+            hwid = win_dev.get_hwids(session, device_name, devcon_folder, login_timeout)[0]
+            session.cmd(params["msi_cmd"] % (hwid, 0))
+            session = vm.reboot(session=session)
+            error_context.context("Check %s's irq number" % device_name, logging.info)
+            get_irq_cmd = params["get_irq_cmd"] % (devcon_folder, hwid)
+            irq_list = re.findall(r':\s+(\d+)', session.cmd_output(get_irq_cmd), re.M)
+            if not irq_list:
+                test.error("device %s's irq checked fail" % device_name)
+            irq_nums = len(irq_list)
+            for irq_symbol in (ctypes.c_int32(int(irq)).value for irq in irq_list):
+                if not irq_nums == 1 and irq_symbol > 0:
+                    test.fail("%s's irq is not correct." % device_name)
+
+    # prepare test data
+    guest_path = (tmp_dir + "src-%s" % utils_misc.generate_random_string(8))
+    host_path = os.path.join(test.tmpdir, "tmp-%s" %
+                             utils_misc.generate_random_string(8))
+    logging.info("Test setup: Creating %dMB file on host", filesize)
+    process.run(dd_cmd % (host_path, filesize), shell=True)
+
+    try:
+        src_md5 = crypto.hash_file(host_path, algorithm="md5")
+        logging.info("md5 value of data from src: %s" % src_md5,
+                     logging.info)
+        # transfer data
+        error_context.context("Transfer data from host to %s" % vm.name,
+                              logging.info)
+        vm.copy_files_to(host_path, guest_path)
+        dst_md5 = get_file_md5sum(guest_path, session,
+                                  timeout=file_md5_check_timeout)
+        logging.info("md5 value of data in %s: %s" % (vm.name, dst_md5),
```

Comment:
same here

Reply:
I have update it, please help me review it again.thanks.